pkgs/tools/filesystems/ceph/default.nix

```diff
@@ -167,6 +167,18 @@ let
   # Watch out for python <> boost compatibility
   python = python310.override {
     packageOverrides = self: super: {
+      # Ceph does not support `cryptography` > 40 yet:
+      #     https://github.com/NixOS/nixpkgs/pull/281858#issuecomment-1899358602
+      # We pin the older `cryptography_40` here;
+      # this also forces us to pin an older `pyopenssl` because the current one
+      # is not compatible with older `cryptography`, see:
+      #     https://github.com/pyca/pyopenssl/blob/d9752e44127ba36041b045417af8a0bf16ec4f1e/CHANGELOG.rst#2320-2023-05-30
+      cryptography = super.cryptography_40;
+      pyopenssl = super.pyopenssl_23_1;
```

Comment:
```suggestion
      cryptography = super.cryptography.overridePythonAttrs (old: {
        version = "40.0.1";
        ...
      });
      pyopenssl = super.pyopenssl.overridePythonAttrs (old: {
        ...
      });
```

Reply:
I've implemented it now, but I think there's a risk that this might break more often when the overridden package is updated (e.g. the list of dependencies changes). Let's see how it goes.